src/AppInstallerCLICore/Workflows/ConfigurationFlow.cpp

```diff
@@ -959,6 +961,87 @@ namespace AppInstaller::CLI::Workflow
+            {
+                return Settings::ConfigurationAllowedZonesOptions::UntrustedSites;
+            }
+
+            return static_cast<Settings::ConfigurationAllowedZonesOptions>(dwZone);
+        }
+
+        // Validate group policy for a given zone.
+        bool IsBlockedByGroupPolicy(Execution::Context& context, const Settings::ConfigurationAllowedZonesOptions zone)
+        {
+            auto configurationPolicies = Settings::GroupPolicies().GetValue<Settings::ValuePolicy::ConfigurationAllowedZones>();
+            if (!configurationPolicies.has_value())
+            {
+                AICLI_LOG(Config, Warning, << "ConfigurationAllowedZones policy is not set");
+                return false;
+            }
+
+            if (configurationPolicies->find(zone) == configurationPolicies->end())
+            {
+                AICLI_LOG(Config, Warning, << "Configuration is not configured in the zone " << zone);
+                return false;
+            }
+
+            auto isAllowed = configurationPolicies->at(zone);
+            if(!isAllowed)
+            {
+                context.Reporter.Error() << "Configuration is disabled for Zone: " << zone << std::endl;
+                return true;
+            }
+
+            AICLI_LOG(Config, Info, << "Configuration is configured in zone " << zone << " with value " << (isAllowed ? "allowed" : "blocked"));
+            return false;
+        }
+
+        // Validate smart screen for a given url.
+        bool IsBlockedBySmartScreen(Execution::Context& context, const std::string& url)
+        {
+            auto response = AppInstaller::UriValidation::UriValidation(url);
+            switch (response.Decision())
+            {
+            case AppInstaller::UriValidation::UriValidationDecision::Block:
+                context.Reporter.Error() << std::endl << "Blocked by smart screen" << std::endl << "Feedback: " << response.Feedback() << std::endl;
+                return true;
+            case AppInstaller::UriValidation::UriValidationDecision::Allow:
+            default:
+                return false;
+            }
+        }
+
+        bool IsSmartScreenRequired(Settings::ConfigurationAllowedZonesOptions zone)
+        {
+            return zone == Settings::ConfigurationAllowedZonesOptions::Internet
+                || zone == Settings::ConfigurationAllowedZonesOptions::UntrustedSites;
+        }
+
+        // Evaluate a given uri for configuration.
+        HRESULT EvaluateUri(Execution::Context& context, const std::string& uri)
+        {
+            auto zone = GetUriZone(uri);
+            if(IsBlockedByGroupPolicy(context, zone))
+            {
+                return APPINSTALLER_CLI_ERROR_BLOCKED_BY_POLICY;
+            }
+
+            if (IsSmartScreenRequired(zone) && IsBlockedBySmartScreen(context, uri))
+            {
+                return APPINSTALLER_CLI_ERROR_SOURCE_NOT_SECURE;

```

Comment:
@JohnMcPMS, any suggestion if instead I should create a new error code here? e.g. 'blocked by smartscreen'?

Reply:
You should not use that error code.  I would also prefer to be able to differentiate between a policy-based block and one based on inspection of the URI by a service.  The relevant party to take things up with is different between these two cases.

Given that, I would suggest two new error codes: "blocked by policy" and "blocked by reputation service".

Reply:
Great! Yes, that error code was awkward here 🙂 I will add a new one, thank you. For the "blocked by policy" one, I am using `APPINSTALLER_CLI_ERROR_BLOCKED_BY_POLICY` in the line above. Is your suggestion to create another one like APPINSTALLER_CLI_ERROR_URL_ZONE_BLOCKED_BY_POLICY?

Reply:
No, using that existing one is fine.  I didn't look into the change to see that you were already doing that.